reco_utils/recommender/deeprec/models/base_model.py

```diff
@@ -483,6 +486,46 @@ def run_eval(self, filename):
             labels.extend(np.reshape(step_labels, -1))
         res = cal_metric(labels, preds, self.hparams.metrics)
         return res
+    
+    def run_eval_topN(self, filename, hparams):
+        load_sess = self.sess
+        preds = []
+        labels = []
+        for batch_data_input in self.iterator.load_data_from_file(filename):
+            _, _, step_pred, step_labels = self.eval(load_sess, batch_data_input)
+            preds.extend(np.reshape(step_pred, -1))
+            labels.extend(np.reshape(step_labels, -1))
+            
+        
+        user_data = pd.read_csv (hparams.user_item_file, sep='\t')[['userID','itemID']]
```

Comment:
there's a space between read_csv and (hparams, that will cause an error. is this function tested under any of the notebook tests?